examples/waspleau/Notes.md

```diff
@@ -0,0 +1,198 @@
+| --------------------------------------------------------------- |
+
+## 2) Axios version pinning
+
+```
+wasp: Error: Dependency conflict for user npm dependency (axios, ^0.21.1): Version must be set to the exactly the same version as the one wasp is using: ^0.21.1
+CallStack (from HasCallStack):
+  error, called at src/Generator/WebAppGenerator.hs:73:31 in waspc-0.2.2.2-B3XOTY4be7eFFZyFqJsQfH:Generator.WebAppGenerator
+```
+
+## 3) Dockerfile keeps getting overwritten. Any way to add to it? Do we want to use Docker Compose to allow for additional services like Redis?
+
+## 4) Dockerfile python stale/heroku container does not support M1/aarch64 [`heroku container:push` did not work]
+
+### First error: python missing (probably for all architectures)
+
+```
+ => [server-production 1/5] WORKDIR /app                                                                          0.0s
+ => ERROR [server-builder 1/6] RUN apk add --no-cache build-base libtool autoconf automake python                 0.5s
+------
+ > [server-builder 1/6] RUN apk add --no-cache build-base libtool autoconf automake python:
+#7 0.194 fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/aarch64/APKINDEX.tar.gz
+#7 0.366 fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/aarch64/APKINDEX.tar.gz
+#7 0.502 ERROR: unable to select packages:
+#7 0.517   python (no such package):
+#7 0.517     required by: world[python]
+------
+executor failed running [/bin/sh -c apk add --no-cache build-base libtool autoconf automake python]: exit code: 1
+ ▸    Error: docker build exited with Error: 1
+ ```
+
+ Solution was to change base `python` to `python3`.
+
+### Next errors: Prisma openssl related for M1
+
+To bypass all of that, I had to manually build exclusively for x86 first like so: `docker buildx build --platform linux/amd64 . -t web`
+
+I could then run:
+```
+docker tag web registry.heroku.com/waspleau/web
+docker push registry.heroku.com/waspleau/web
+```
+
+Snip to save: ` docker buildx build --platform linux/amd64 . -t web && docker tag web registry.heroku.com/waspleau/web && docker push registry.heroku.com/waspleau/web && heroku container:release --app waspleau web`
+
+## 5) When I pushed to Heroku I had a migration_lock.toml issue
+
+```
+2021-12-16T02:00:14.338320+00:00 app[web.1]: Prisma schema loaded from ../db/schema.prisma
+2021-12-16T02:00:14.366442+00:00 app[web.1]: Datasource "db": PostgreSQL database "dbne5egge1dhr8", schema "public" at "ec2-184-73-25-2.compute-1.amazonaws.com:5432"
+2021-12-16T02:00:14.564581+00:00 app[web.1]: Error: P3014
+2021-12-16T02:00:14.564591+00:00 app[web.1]:
+2021-12-16T02:00:14.564592+00:00 app[web.1]: The datasource provider `postgresql` specified in your schema does not match the one specified in the migration_lock.toml, `sqlite`. Please remove your current migration directory and start a new migration history with prisma migrate dev. Read more: https://pris.ly/d/migrate-provider-switch
```

Comment:
Aha, this is because you switched from sqlite to postgresql -> yeah, once you do that you need to drop the existing migrations, that is by design (by Prisma). I guess we didn't mention this in the docs anywhere?

Reply:
We do have it there: https://wasp-lang.dev/docs/language/basic-elements#migrating-from-sqlite-to-postgresql I will likely remove this part from my notes.

Reply:
good info on missing that, probably will happen to others too. It might be cool in the future to have a mechanism that detects dev switched so could e.g. warn them through CLI. Def an over-optimisation for now, but it was just a thought :D.

Reply:
Others may read better than I do 🤓  lol but yes, I agree it would be great if in the future it detected the switch of database types and warned or did it for you 🚀